### PR TITLE
ARROW-2000: [Plasma] Deduplicate file descriptors when replying to GetRequest.

### PR DESCRIPTION
**Pull request by robertnishihara**

### CODE REVIEW

**Comment by robertnishihara** [comment]
cc @pcmoritz

File: cpp/src/plasma/format/plasma.fbs [hunk 1]
```diff
@@ -156,9 +156,17 @@ table PlasmaGetReply {
...
   plasma_objects: [PlasmaObjectSpec];
-  // The number of elements in both object_ids and plasma_objects arrays must agree.
+  // A list of the file descriptors in the store that correspond to the file
+  // descriptors being sent to the client. The length of this list is the number
+  // of file descriptors that the store will send to the client after this
+  // message.
+  store_file_descriptors: [int];
+  // Size in bytes of the segment for each store file descriptor (needed to call
+  // mmap). This list must have the same length as store_file_descriptors.
+  mmap_sizes: [long];
```
Comment (robertnishihara) [inline cpp/src/plasma/format/plasma.fbs]: Consider getting rid of this info from the `PlasmaObjectSpec`.

File: cpp/src/plasma/protocol.cc [hunk 2]
```diff
@@ -76,27 +76,33 @@ Status ReadCreateRequest(uint8_t* data, size_t size, ObjectID* object_id,
...
 Status ReadCreateReply(uint8_t* data, size_t size, ObjectID* object_id,
-                       PlasmaObject* object) {
+                       PlasmaObject* object, int* store_fd, int64_t* mmap_size) {
   DCHECK(data);
   auto message = flatbuffers::GetRoot<PlasmaCreateReply>(data);
   DCHECK(verify_flatbuffer(message, data, size));
   *object_id = ObjectID::from_binary(message->object_id()->str());
   object->handle.store_fd = message->plasma_object()->segment_index();
-  object->handle.mmap_size = message->plasma_object()->mmap_size();
+  // TODO(rkn): Remove mmap_size from the plasma store's object table.
```
Comment (pcmoritz) [inline cpp/src/plasma/protocol.cc]: Can we do this as part of this PR? How hard is it?

**Review by pcmoritz — APPROVED** [review]
Looks good, but please remove mmap_size from the object table

**Pull request closed without merging**